lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -35,14 +56,16 @@ def get_problem_grade_distribution(course_id):
       'total_student_count' where the key is problem 'module_id' and the value is number of students
         attempting the problem
     """
+    non_student_list = get_non_student_list(course_id)
 
     # Aggregate query on studentmodule table for grade data for all problems in course
-    db_query = models.StudentModule.objects.filter(
+    db_query = models.StudentModule.objects.select_related('student').filter(
```

Comment:
You are right! I think I was playing around with django ORM trying to learn what different things it can do and left it there.  There are 3 places that I could just check for student_id.  I will fix them.